src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -276,10 +271,10 @@ static grpc_channel *client_channel_factory_create_channel(
                                               GRPC_CLIENT_CHANNEL, NULL);
   grpc_channel_args_destroy(final_args);
 
-  grpc_resolver *resolver = grpc_resolver_create(target, &f->base);
+  grpc_resolver *resolver = grpc_resolver_create(target);
   if (resolver != NULL) {
-    grpc_client_channel_set_resolver(
-        exec_ctx, grpc_channel_get_channel_stack(channel), resolver);
+    grpc_client_channel_set_resolver_and_client_channel_factory(
```

Comment:
What happens when the next factory object needs to be handed in?

grpc_client_channel_set_resolver_and_client_channel_factory_and_load_balancing_factory_and_retry_policy_factory(...);